src/shogun/features/SNPFeatures.cpp

```diff
@@ -401,8 +401,8 @@ SGMatrix<float64_t> SNPFeatures::get_histogram(bool normalize)
 		}
 	}
 	SG_FREE(h_normalizer);
-
-	return SGMatrix<float64_t>(h, nsym, string_length/2);
+	SGMatrix<float64_t>hMatrix(h, nsym, string_length/2,false);
```

Comment:
spaces plz

Reply:
note that this is actually just putting one line up the matrix creation.... but the underlying memory chunk `h` is still created by SG_MALLOC/CALLOC see https://github.com/shogun-toolbox/shogun/blob/9bd877047b31461a05ef97f489a17b32c3d61427/src/shogun/features/SNPFeatures.cpp#L354

you should just create there:
```
SGMatrix<float64_t> h(nsym, string_length/2,false);
```
instead of the calloc, and return `h` itself